server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/TaskApplication.java

```diff
@@ -88,15 +88,17 @@ private void init(String[] args) {
         context = JobContextProviderFactory.create().provide();
         // 4 step: trace taskId in log4j2 context
         trace(context.getJobIdentity().getId());
-        // 5 step: set log4j2.xml
+        // 5 step: set log path in system properties
+        setLogPathSysProperty();
+        // 6 step: set log4j2.xml
         setLog4JConfigXml();
 
         log.info("Task executor start info, ip={}, port={}, runMode={}, taskId={}, logPath={}, userId={}.",
                 SystemUtils.getLocalIpAddress(),
                 SystemUtils.getEnvOrProperty(JobEnvKeyConstants.ODC_EXECUTOR_PORT),
                 SystemUtils.getEnvOrProperty(JobEnvKeyConstants.ODC_TASK_RUN_MODE),
                 context.getJobIdentity().getId(),
-                SystemUtils.getEnvOrProperty(JobEnvKeyConstants.ODC_LOG_DIRECTORY),
+                System.getProperty(JobEnvKeyConstants.ODC_LOG_DIRECTORY),
```

Comment:
Why can not use `getEnvOrProperty()`?

Reply:
in 5 step: setLogPathSysProperty(); `ODC_LOG_DIRECTORY` will be set in system properties, we will use system properties in log4j2-task.xml